packages/core/src/interpreter.ts

```diff
@@ -150,7 +142,12 @@ export class Actor<TLogic extends AnyActorLogic>
     const { clock, logger, parent, syncSnapshot, id, systemId, inspect } =
       resolvedOptions;
 
-    this.system = parent?.system ?? createSystem(this);
+    this.system =
+      parent?.system ??
+      createSystem(this, {
+        clock,
+        snapshot: (options as any)?.snapshot?.['xstate.system']
```

Comment:
for the root we probably should "split" the `options.snapshot` so `restoreSnapshot` of the root logic doesn't receive this `xstate.system` key there. We need to contain all of this attaching/detaching of the `xstate.system` magical thing in the Actor

Reply:
This part is removed